### DISCUSSION TITLE
How to join a domain?

### DISCUSSION THREAD

**Discussion by flossgeek** [cockpit-project/cockpit#16635] [Q&A] [answered]
Could someone explain me how one would join a domain? I am trying to connect the cockpit UI to a subdomain, but cockpit can't contact it. I also can't find clear documentation about this anywhere, so that would be a great addition :)

**Comment by martinpitt** [comment]
I'm afraid we need some more information. What does `realm discover` show? Can it "see" the domain that you want to join? You can't join a "subdomain" of that -- you can join with specifying a *DNS* subdomain, and cockpit/realmd can then ask DNS about whether there's a matching FreeIPA domain; but that will still join the original FreeIPA domain.

**Reply by flossgeek** [reply]
Thanks for the fast reply! `realm discover "domain"` times out and gives a notice that the realm is not found, both with the original domain and the subdomain. 
What DNS configurations do I need for the DNS subdomain?

**Comment by martinpitt** [comment]
What does `realm discover` show (without arguments)? Normally your FreeIPA and DNS should be set up so that it can be auto-discovered through DNS. Apparently DNS or possibly firewall etc. are broken so that your machine can't talk to the FreeIPA server?

Anything useful in `sudo journalctl -f` while you run the discover command?

**Reply by flossgeek** [reply]
Without arguments it outputs `realm: No default realm discovered` and `* No default domain received via DHCP or given by hostname`. The domain does not yet have a connection to the server cockpit runs on though. Cockpit now runs on a public IP and I want to substitute that with a subdomain, so I hoped that "Join Domain" in the system configuration could fulfill that purpose.

**Comment by martinpitt** [comment, marked as answer]
> The domain does not yet have a connection to the server cockpit runs on though

I'm not entirely sure what you mean, but the machine you want to join must of course have a network connection to the FreeIPA server, and already be in the same DNS domain.

**Reply by flossgeek** [reply]
Thanks for explaining. This makes me realize that it's probably easier to do it the other way around: pointing the domain to the IP with an A record. Thanks for your time!
